hudi-common/src/test/java/org/apache/hudi/avro/TestMercifulJsonConverter.java

```diff
@@ -99,4 +106,44 @@ public void conversionWithFieldNameAliases() throws IOException {
 
     Assertions.assertEquals(rec, CONVERTER.convert(json, sanitizedSchema));
   }
+
+  @Test
+  public void testConvertNumberToFixed() throws IOException {
+    String testSchemaStr = "{\"type\": \"record\",\"name\": \"test_record\",\"namespace\": \"test_namespace\",\"fields\": "
+        + "[{\"name\": \"decimal_field\",\"type\": [\"null\",{\"type\": \"fixed\",\"name\": \"fixed\",\"namespace\": \"test_namespace.decimal_field\",\"size\": 9,"
+        + "    \"logicalType\": \"decimal\",\"precision\": 20,\"scale\": 0}],\"default\": null},"
+        + "{\"name\": \"decimal2_field\",\"type\": [\"null\",{\"type\": \"fixed\",\"name\": \"fixed\",\"namespace\": \"test_namespace.decimal2_field\",\"size\": 9,"
+        + "    \"logicalType\": \"decimal\",\"precision\": 20,\"scale\": 2}],\"default\": null},"
+        + "{\"name\": \"decimal3_field\",\"type\": [\"null\",{\"type\": \"fixed\",\"name\": \"fixed\",\"namespace\": \"test_namespace.decimal3_field\",\"size\": 9,"
+        + "    \"logicalType\": \"decimal\",\"precision\": 20,\"scale\": 2}],\"default\": null},"
+        + "{\"name\": \"int_field\",\"type\": [\"null\",\"int\"],\"default\": null},"
+        + "{\"name\": \"long_field\",\"type\": [\"null\",\"long\"],\"default\": null},"
+        + "{\"name\": \"string_field\",\"type\": [\"null\",\"string\"],\"default\": null}]}";
+    Schema schema = Schema.parse(testSchemaStr);
+
+    String testValueStr = "{\n"
+        + "    \"decimal_field\": 1720623716,\n"
```

Comment:
the schema type is `Fixed`, why use `Number` value for test?

Reply:
Because avro store Decimal type as Fixed type internally

Reply:
the decimal type is `logicalType`, data type is Fixed, which is bytes, the logical type `decimal` I think in fixed is used to conver `bytes` to `decimal`, not conver `number` to `decimal`.